WIP: UPSTREAM: 113971: wire context in rbac

s-urbaniak opened the pull request:
This wires context into rule resolving in the authorization subsystem.

This is needed to fix https://github.com/kcp-dev/kcp/issues/1340

ncdc commented:
Wrong title for PR?

s-urbaniak commented:
oh dear, yes thank you :D

ncdc commented on plugin/pkg/auth/authorizer/rbac/subject_locator.go:
```diff
@@ -66,12 +70,12 @@ func (r *SubjectAccessEvaluator) AllowedSubjects(requestAttributes authorizer.At
 	errorlist := []error{}
 
-	if clusterRoleBindings, err := r.clusterRoleBindingLister.ListClusterRoleBindings(); err != nil {
+	if clusterRoleBindings, err := r.clusterRoleBindingLister.ListClusterRoleBindings(genericapirequest.NewContext()); err != nil {
```
It's ok to have an empty context here?

s-urbaniak replied:
I actually wired a context in the subject locator. An empty context will cause a panic down below in the stack.

s-urbaniak committed:
pkg/registry/rbac/*,plugin/pkg/auth/authorizer/*: wire context

Currently, RBAC registry adapters create a request context on the fly
but don't reuse the existing request context.

This fixes it by wiring the context.

s-urbaniak commented:
xref https://github.com/kubernetes/kubernetes/pull/113971

s-urbaniak commented:
upstream contribution is submitted https://github.com/kubernetes/kubernetes/pull/113971

stevekuznetsov commented:
/retitle UPSTREAM: 113971: wire context in rbac
